tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -3288,8 +3288,8 @@ mlir::Value MLIRScanner::GetAddressOfDerivedClass(
 
     // Add the offset.
 
-    mlir::Type nt = getMLIRType(
-        Glob.CGM.getContext().getLValueReferenceType(Base->getType()));
+    mlir::Type nt = getMLIRType(Glob.CGM.getContext().getLValueReferenceType(
+        QualType(RD->getTypeForDecl(), 0)));
```

Comment:
This change was try and error and starting from the actual clang type declaration seems to work. Let me know if this does not make sense.

Reply:
Let me double check then. I just realize that toBase and Base are indexed differently. I probably need to get to a Qualified type from Base.

Reply:
The thing I'd add as a check here, is something with an offset (e.g. make multiple base classes so the cast to a base wouldn't be 0)

Reply:
Yes that makes sense! Will do that.

Reply:
I made a more complex test that also uses the struct abi and multiple inheritance. I think the result looks correct? Also I looked through the code again and the nt variable I changed seems to be used to convert to the derived type. Using RD from the toBase array thus seems correct.

Reply:
Okay seems reasonable to me. One more ask, since this is a for loop multiple class cast, can you also include a cast to an ancestor (but not parent) class (i.e. C derives B, B derives A).

Otherwise LGTM.

Reply:
Sure will look into that tomorrow.